src/Promotion/Action/UnitsFixedPricePromotionActionCommand.php

```diff
@@ -0,0 +1,169 @@
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class UnitsFixedPricePromotionActionCommand extends UnitDiscountPromotionActionCommand
+{
+    public const TYPE = 'units_fixed_price';
+
+    /** @var FilterInterface */
+    private $priceRangeFilter;
+
+    /** @var FilterInterface */
+    private $taxonFilter;
+
+    /** @var FilterInterface */
+    private $productFilter;
+
+    /** @var MostExpensiveFirstDistributorInterface */
+    private $distributor;
+
+    public function __construct(
+        FactoryInterface $adjustmentFactory,
+        FilterInterface $priceRangeFilter,
+        FilterInterface $taxonFilter,
+        FilterInterface $productFilter,
+        MostExpensiveFirstDistributorInterface $distributor
+    ) {
+        parent::__construct($adjustmentFactory);
+
+        $this->priceRangeFilter = $priceRangeFilter;
+        $this->taxonFilter = $taxonFilter;
+        $this->productFilter = $productFilter;
+        $this->distributor = $distributor;
+    }
+
+    public function execute(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion): bool
+    {
+        if (!$subject instanceof OrderInterface) {
+            throw new UnexpectedTypeException($subject, OrderInterface::class);
+        }
+
+        $channel = $subject->getChannel();
+        if (null === $channel) {
+            return false;
+        }
+
+        $channelCode = $channel->getCode();
+        if (!isset($configuration[$channelCode])) {
+            return false;
+        }
+
+        if (empty($configuration[$channelCode]['amount']) || empty($configuration[$channelCode]['itemsAmount'])) {
+            return false;
+        }
+        $amount = $configuration[$channelCode]['amount'];
+        $itemsAmount = $configuration[$channelCode]['itemsAmount'];
+
+        $filteredItems = $this->getFilteredItems($subject, $configuration[$channelCode]);
+        if (empty($filteredItems)) {
+            return false;
+        }
+
+        $amount = $this->getExpectedTotal($amount, $itemsAmount, $filteredItems);
+        // If the items total is already lower than expected discounted amount, return there
+        $currentTotal = $this->getEligibleItemsTotal($filteredItems);
+        if ($currentTotal <= $amount) {
+            return false;
+        }
+
+        $discountAmount = $currentTotal - $amount;
+        $units = $this->getUnits($filteredItems);
+        $unitsTotals = [];
+        foreach ($units as $unit) {
+            $unitsTotals[] = $unit->getTotal();
+        }
+
+        $splitPromotion = $this->distributor->distribute($unitsTotals, $discountAmount);
+
+        foreach ($units as $key => $unit) {
+            $this->setUnitAdjustments($unit, $splitPromotion[$key], $promotion);
+        }
```

Comment:
Is this the way Sylius does it in their distributors?

Reply:
Yes it is. I can just refactor a bit